session_rpcserver.go

```diff
@@ -1213,6 +1213,18 @@ func (s *sessionRpcServer) AddAutopilotSession(ctx context.Context,
 		return nil, fmt.Errorf("error creating new session: %v", err)
 	}
 
+	// If we tried to link to a previous session, we delete the newly
+	// created session in the case of errors to avoid having non-revoked
+	// sessions lying around.
+	fail := func(err error) error {
```

Comment:
Nit: Not sure we need a separate lambda function for this when it's only being called once?